recipes/CommonVoice/ASR/transformer/train.py

```diff
@@ -268,14 +269,12 @@ def on_fit_start(self):
         # Wrap modules with parallel backend after jit
         self._wrap_distributed()
 
-        # Initialize optimizers after parameters are configured
-        self.init_optimizers()
-
-        # Load latest checkpoint to check to current epoch number
         if self.checkpointer is not None:
             self.checkpointer.recover_if_possible()
 
-        # if the model is resumed from stage two, reinitialize the optimizer
+        # Initialize optimizers after parameters are configured
+        self.init_optimizers()
+
         current_epoch = self.hparams.epoch_counter.current
         if current_epoch > self.hparams.stage_one_epochs:
             self.optimizer = self.hparams.SGD(self.modules.parameters())
```

Comment:
I think it would be great to add a logging message on line 280 saying that you are loading the SGD opt (so that the user can be assured that everything works as intended)